### PR TITLE
webp: build with CMake

### LINKED ISSUE #126823: webp doesn't install CMake configs
### `brew gist-logs <formula>` link OR `brew config` AND `brew doctor` output

```shell
brew config
HOMEBREW_VERSION: 4.0.10
ORIGIN: https://github.com/Homebrew/brew
HEAD: cf6614b50eac1c9f4bf386dc6989f0d443148df8
Last commit: 4 hours ago
Core tap JSON: 27 Mar 17:27 UTC
HOMEBREW_PREFIX: /home/linuxbrew/.linuxbrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_DISPLAY: :0
HOMEBREW_MAKE_JOBS: 4
Homebrew Ruby: 2.6.8 => /home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.8_1/bin/ruby
CPU: quad-core 64-bit bulldozer
Clang: N/A
Git: 2.17.1 => /usr/bin/git
Curl: 7.58.0 => /usr/bin/curl
Kernel: Linux 4.15.0-111-generic x86_64 GNU/Linux
OS: Ubuntu 18.04.4 LTS (bionic)
Host glibc: 2.27
/usr/bin/gcc: 7.5.0
/usr/bin/ruby: 2.5.1
glibc: N/A
gcc@11: N/A
gcc: N/A
xorg: N/A
```

```
$ brew doctor
Your system is ready to brew.
```
```


### Verification

- [X] My "`brew doctor` output" says `Your system is ready to brew.` and am still able to reproduce my issue.
- [X] I ran `brew update` and am still able to reproduce my issue.
- [X] I have resolved all warnings from `brew doctor` and that did not fix my problem.

### What were you trying to do (and why)?

`brew install webp`

### What happened (include all command output)?

brew installed webp without CMake configs. webp client applications cannot use `find_package(WebP)`

### What did you expect to happen?

webp ships Cmake configs in the source archive. I expect them to be installed along with pkg-config files.

### Step-by-step reproduction instructions (by running `brew` commands)

```shell
brew install webp
brew ls -v webp
```
```

### PR DESCRIPTION
**Pull request by ZhongRuoyu**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

Resolves #126823.

### CODE REVIEW

File: Formula/webp.rb [hunk 1]
```diff
@@ -15,27 +16,16 @@ class Webp < Formula
...
   def install
-    system "./autogen.sh" if build.head?
-    system "./configure", "--prefix=#{prefix}",
-                          "--disable-dependency-tracking",
-                          "--disable-gl",
-                          "--enable-libwebpdecoder",
-                          "--enable-libwebpdemux",
-                          "--enable-libwebpmux"
-    system "make", "install"
+    system "cmake", "-S", ".", "-B", "build", *std_cmake_args
```
Comment (carlocab) [inline Formula/webp.rb:26]: Probably want `BUILD_SHARED_LIBS=ON`. We may need `CMAKE_INSTALL_RPATH` too.

**Comment by p-linnane** [comment]
Ran out of disk space on Linux. Retrying with self-hosted.

**Comment by ZhongRuoyu** [comment]
```
==> brew linkage --test qt@5
==> FAILED
Full linkage --test qt@5 output
  Missing libraries:
    unexpected (libclang.so.13)
Truncated linkage --test qt@5 output
  Error: Missing libraries:
    unexpected (libclang.so.13)
```

**Review by carlocab — APPROVED** [review]
Linux failure should be unrelated. Not really sure what the problem is, but it's definitely not because of this change.

**Pull request merged**